src/main/java/it/gov/pagopa/reward/service/synchronous/RewardTrxSynchronousApiServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+import it.gov.pagopa.reward.service.reward.RewardContextHolderService;
+import it.gov.pagopa.reward.service.reward.evaluate.InitiativesEvaluatorFacadeService;
+import it.gov.pagopa.reward.utils.RewardConstants;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.Map;
+
+@Service
+@Slf4j
+public class RewardTrxSynchronousApiServiceImpl implements RewardTrxSynchronousApiService {
+    private final RewardContextHolderService rewardContextHolderService;
+    private final OnboardedInitiativesService onboardedInitiativesService;
+    private final InitiativesEvaluatorFacadeService initiativesEvaluatorFacadeService;
+
+    private final UserInitiativeCountersRepository userInitiativeCountersRepository;
+    private final SynchronousTransactionRequestDTOt2TrxDtoOrResponseMapper trxPreviewRequest2TransactionDtoMapper;
+    private final RewardTransaction2SynchronousTransactionResponseDTOMapper rewardTransaction2SynchronousTransactionResponseDTOMapper;
+
+    public RewardTrxSynchronousApiServiceImpl(RewardContextHolderService rewardContextHolderService, OnboardedInitiativesService onboardedInitiativesService, InitiativesEvaluatorFacadeService initiativesEvaluatorFacadeService, UserInitiativeCountersRepository userInitiativeCountersRepository, SynchronousTransactionRequestDTOt2TrxDtoOrResponseMapper trxPreviewRequest2TransactionDtoMapper, RewardTransaction2SynchronousTransactionResponseDTOMapper rewardTransaction2SynchronousTransactionResponseDTOMapper) {
+        this.rewardContextHolderService = rewardContextHolderService;
+        this.onboardedInitiativesService = onboardedInitiativesService;
+        this.initiativesEvaluatorFacadeService = initiativesEvaluatorFacadeService;
+        this.userInitiativeCountersRepository = userInitiativeCountersRepository;
+        this.trxPreviewRequest2TransactionDtoMapper = trxPreviewRequest2TransactionDtoMapper;
+        this.rewardTransaction2SynchronousTransactionResponseDTOMapper = rewardTransaction2SynchronousTransactionResponseDTOMapper;
+    }
+
+    @Override
+    public Mono<SynchronousTransactionResponseDTO> previewTransaction(SynchronousTransactionRequestDTO trxPreviewRequest, String initiativeId) {
+        log.trace("[REWARD] Starting reward preview calculation for transaction {}", trxPreviewRequest.getTransactionId());
+        TransactionDTO trxDTO = trxPreviewRequest2TransactionDtoMapper.apply(trxPreviewRequest);
+
+        return checkInitiative(trxPreviewRequest, initiativeId)
+                .flatMap(b -> checkOnboarded(trxPreviewRequest, trxDTO, initiativeId))
+                .flatMap(b -> userInitiativeCountersRepository.findById(UserInitiativeCounters.buildId(trxDTO.getUserId(),initiativeId)))
+                .switchIfEmpty(Mono.just(UserInitiativeCounters.builder(trxPreviewRequest.getUserId(), initiativeId).build()))
```

Comment:
just call the constructor
```suggestion
                .switchIfEmpty(Mono.just(new UserInitiativeCounters(trxPreviewRequest.getUserId(), initiativeId)))
```